src/linked_list.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Create a singly linked list."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Create a node to store data."""
+
+    def __init__(self, value, pointer=None):
+        """Create an instance of a Node."""
+        self.value = value
+        self.pointer = pointer
+
+
+class LinkedList(object):
+    """Create a link list class to store strings of data."""
+
+    def __init__(self, value_list=None):
+        """Create an instance of a singly-linked list."""
+        self.length = 0
+        self.header = None
+        if hasattr(value_list, '__iter__'):
```

Comment:
The assignment is quite explicit that the constructor should accept an iterable.  You don't need to account for, nor should you handle, passing single, non-iterable values.